src/server/pfs/server/egress.go

```diff
@@ -0,0 +1,99 @@
+package server
+
+import (
+	"io"
+	"os"
+	"strings"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/miscutil"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
+	"github.com/pachyderm/pachyderm/v2/src/internal/sdata"
+	"github.com/pachyderm/pachyderm/v2/src/internal/storage/fileset"
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+	"golang.org/x/net/context"
+)
+
+type egressResult map[string]int64
+
+func getEgressPassword() (string, error) {
+	const passwordEnvar = "PACHYDERM_SQL_PASSWORD" // TODO move this to a package for sharing
+	password, ok := os.LookupEnv(passwordEnvar)
+	if !ok {
+		return "", errors.Errorf("must set %v", passwordEnvar)
+	}
+	return password, nil
```

Comment:
should we enforce this in `ValidateInput` during pipeline creation? Either the user needs to put the secret field in the egress, or we hope they provide it some way, I feel like it makes some things simpler if we assume we're the ones creating it